client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -332,13 +334,16 @@ private Map<String, SplitResult> getTreatmentsWithConfigInternal(String matching
                 _log.warn("The sets are not in flagSetsFilter config");
                 return new HashMap<>();
             }
-            featureFlagNames = getAllFlags(cleanFlagSets);
+            featureFlagNames = new ArrayList<>();
         } else if (featureFlagNames == null) {
             _log.error(String.format("%s: featureFlagNames must be a non-empty array", methodEnum.getMethod()));
             return new HashMap<>();
         }
         try {
             checkSDKReady(methodEnum, featureFlagNames);
```

Comment:
why are you not checking if the SDK is ready in the first line of this method? you are doing a lot of operations before checking the status, I think you can improve the "performance" doing the validations in the beginning.

ps: the method is really hard to read/understand because it has many `if else` conditions and to many lines.

Reply:
I moved set validation to another method.